src/scripts.js

```diff
@@ -1,168 +1,167 @@
   greetUser();
 }
 
-function viewFavorites() {
+const greetUser = () => {
+  welcomeMessage.innerText = `Welcome ${domUser.name}!`;
+};
+
+const populateCards = (recipes) => {
+  cardArea.innerHTML = '';
   if (cardArea.classList.contains('all')) {
-    cardArea.classList.remove('all')
+    cardArea.classList.remove('all') // what does this all do?
   }
-  if (!user.favoriteRecipes.length) {
-    favButton.innerHTML = 'You have no favorites!';
-    populateCards(cookbook.recipes);
-    return
-  } else {
-    favButton.innerHTML = 'Refresh Favorites'
-    cardArea.innerHTML = '';
-    user.favoriteRecipes.forEach(recipe => {
-      cardArea.insertAdjacentHTML('afterbegin', `<div id='${recipe.id}'
-      class='card'>
-      <header id='${recipe.id}' class='card-header'>
-      <label for='add-button' class='hidden'>Click to add recipe</label>
-      <button id='${recipe.id}' aria-label='add-button' class='add-button card-button'>
-      <img id='${recipe.id}' class='add'
-      src='https://image.flaticon.com/icons/svg/32/32339.svg' alt='Add to
-      recipes to cook'></button>
-      <label for='favorite-button' class='hidden'>Click to favorite recipe
-      </label>
-      <button id='${recipe.id}' aria-label='favorite-button' class='favorite favorite-active card-button'>
-      </button></header>
-      <span id='${recipe.id}' class='recipe-name'>${recipe.name}</span>
-      <img id='${recipe.id}' tabindex='0' class='card-picture'
-      src='${recipe.image}' alt='Food from recipe'>
-      </div>`)
+  recipes.forEach(recipe => {
+    cardArea.insertAdjacentHTML('afterbegin', constructCard(recipe))
+  });
+  getFavorites();
+};
+
+const getFavorites = () => {
+  if (domUser.favoriteRecipes.length) {
+    domUser.favoriteRecipes.forEach(recipe => {
+      document.querySelector(`.favorite${recipe.id}`).classList.add('favorite-active')
     })
   }
 }
 
-function greetUser() {
-  const userName = document.querySelector('.user-name');
-  userName.innerHTML =
-  user.name.split(' ')[0] + ' ' + user.name.split(' ')[1][0];
-}
+const constructCard = (recipe, addedClass) => {
+  return `<div id='${recipe.id}'
+  class='card'>
+  <header id='${recipe.id}' class='card-header'>
+  <label for='add-button' class='hidden'>Click to add recipe</label>
+  <button id='${recipe.id}' aria-label='add-button' class='add-button card-button'>
+  <img id='${recipe.id}' class='add'
+  src='https://image.flaticon.com/icons/svg/32/32339.svg' alt='Add to
+  recipes to cook'></button>
+  <label for='favorite-button' class='hidden'>Click to favorite recipe
+  </label>
+  <button id='${recipe.id}' aria-label='favorite-button' class='favorite card-button ${addedClass}'>
+  </button></header>
+  <span id='${recipe.id}' class='recipe-name'>${recipe.name}</span>
+  <img id='${recipe.id}' tabindex='0' class='card-picture'
+  src='${recipe.image}' alt='Food from recipe'>
+  </div>`
+  //The over use of ID's makes me nervous but idk if it's a real issue
```

Comment:
I have the same concerns about the ids.

Reply:
I think the quick fix for this would be to just interpolate recipe.id into the classes rather than ids, even if the element already has one.

Reply:
I think Tim's right, you should be able to have multiple class names on a file and that shouldn't be a very hard fix to update and tweak so every element on a card contains a class name of recipe.id but not have multiple id's being reused.